Sources/StreamChatUI/MessageActionsPopup/ChatMessageActionControl.swift

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// Button for action displayed in `_ChatMessageActionsView`.
+public typealias ChatMessageActionControl = _ChatMessageActionControl<NoExtraData>
+
+/// Button for action displayed in `_ChatMessageActionsView`.
+open class _ChatMessageActionControl<ExtraData: ExtraDataTypes>: _Control, UIConfigProvider {
+    /// The data this view component shows.
+    public var content: ChatMessageActionItem? {
+        didSet { updateContentIfNeeded() }
+    }
+
+    override open var isHighlighted: Bool {
+        didSet {
+            updateContentIfNeeded()
+        }
+    }
+
+    /// `ContainerView` that encapsulates `titleLabel` and `imageView`.
+    public lazy var containerView: ContainerView = ContainerView(alignment: .center)
+        .withoutAutoresizingMaskConstraints
+
+    /// `UILabel` to show `title`.
+    public lazy var titleLabel: UILabel = UILabel()
+
+    /// `UIImageView` to show `image`.
+    public lazy var imageView: UIImageView = UIImageView()
+
+    override public func defaultAppearance() {
+        super.defaultAppearance()
+        backgroundColor = uiConfig.colorPalette.background
+        titleLabel.font = uiConfig.font.body
+        titleLabel.adjustsFontForContentSizeCategory = true
+    }
+
+    override open func setUp() {
+        super.setUp()
+        containerView.isUserInteractionEnabled = false
+        addTarget(self, action: #selector(touchUpInsideHandler(_:)), for: .touchUpInside)
+    }
+    
+    override open func setUpLayout() {
+        super.setUpLayout()
+        
+        addSubview(containerView)
+        embed(containerView, insets: NSDirectionalEdgeInsets(top: 8, leading: 16, bottom: 8, trailing: 16))
+
+        imageView.setContentHuggingPriority(.streamRequire, for: .horizontal)
+        containerView.addArrangedSubview(imageView)
+
+        containerView.addArrangedSubview(titleLabel)
+    }
+
+    override open func tintColorDidChange() {
+        super.tintColorDidChange()
+        
+        updateContentIfNeeded()
+    }
+    
+    override open func updateContent() {
+        let imageTintСolor: UIColor
+        let titleTextColor: UIColor
+
+        if content?.isDestructive == true {
+            imageTintСolor = uiConfig.colorPalette.alert
+            titleTextColor = imageTintСolor
+        } else {
+            imageTintСolor = content?.isPrimary == true ? tintColor : uiConfig.colorPalette.inactiveTint
+            titleTextColor = uiConfig.colorPalette.text
+        }
+
+        titleLabel.text = content?.title
+        if isHighlighted {
+            titleLabel.textColor = uiConfig.colorPalette.highlightedColorForColor(titleTextColor)
+            imageView.image = content?.icon
+                .tinted(with: uiConfig.colorPalette.highlightedColorForColor(imageTintСolor))
+        } else {
```

Comment:
Please check the appearance of the system context menu. The background color should change, too.